### DISCUSSION TITLE
Option to Disable auto change Powerplan

### DISCUSSION THREAD

**Discussion by jurnin** [seerge/g-helper#3119] [Ideas]
Hi I was wondering if you could add a feature to stop Ghelper from auto changing the powerplan incase we want to add our own custom power plan. Or allow you to select your own within the exe

![Screenshot_3](https://github.com/user-attachments/assets/4dcbd183-ac55-4adc-8910-d1952487f339)
![Screenshot_4](https://github.com/user-attachments/assets/65c2e7cc-aa9f-4232-95e8-85f813ee1fb4)

**Comment by seerge** [comment]
@jurnin hello, you can already do that :) [Check readme on how to add it to config](https://github.com/seerge/g-helper/wiki/Power-user-settings#custom-windows-power-plans-with-each-mode)

Dropdown on your screenshot is not power plan btw, it's a **Power Mode**
![image](https://github.com/user-attachments/assets/cd675c6c-a52d-482c-8e8c-54061cb800f0)

**Reply by jurnin** [reply]
Hi yes, I tried adding the custom powerplan within the config but it still changes to Balanced automatically

**Reply by seerge** [reply]
@jurnin then you are not doing it correctly :) you need to quit app before editing and also maintain json structure (i.e. keep commas at the end of each line except last)

Upload your config here

**Comment by jurnin** [comment]
The powerplan I have is a modified version of High Performance, so when setting the powerplan to 1 should still use High performance and it doesnt, it forced Balanced
[config.json](https://github.com/user-attachments/files/17016936/config.json)
![image](https://github.com/user-attachments/assets/b0fb5158-44bb-4e43-855b-be450585ec8f)

**Comment by seerge** [comment]
@jurnin I see that you have set custom power plan only for balanced mode (i.e. middle button in UI)
```
  "scheme_1": "8c5e7fda-e8bf-4a96-9a85-a6e23a8c635c",
```

So it will change power plan when you select it. 

You need to add this setting for every mode you want to customize in that way
```
"scheme_<mode_id>" : <power plan GUID>,
```

**Reply by jurnin** [reply]
Ohh so you mean add like scheme 0, 1, 2 and the same guid for each like this?

![image](https://github.com/user-attachments/assets/a7e9ab91-0207-4f76-bdb5-e0d6de76e786)

**Reply by seerge** [reply]
@jurnin if you want to use **same** High Performance plan for all modes - yes. But I would keep defaults at least for Silent .... (id 2)

**Comment by jurnin** [comment]
I did that and then when I open the program it still forces Balanced

![image](https://github.com/user-attachments/assets/b7e8c924-9a72-4cb9-ac2b-52e14d1ef6c1)
![image](https://github.com/user-attachments/assets/3bc5ac78-5033-4cb4-bd10-094c90c3657d)

**Reply by seerge** [reply]
@jurnin does that power plan exist in your system in a first place? I.e can you set it manually via powercfg command?

Can you upload log from the app?

**Reply by Azzkaryon** [reply]
Hey! He was trying to help me since i am on an asus tuf f15. Can tell me what to send you i am not so advanced in this...

**Reply by Azzkaryon** [reply]
We want to have the high performance power plan with the fans set at max.

**Reply by seerge** [reply]
@Azzkaryon log is located at `%appdata%\GHelper\log.txt`

**Reply by Azzkaryon** [reply]
[log.txt](https://github.com/user-attachments/files/17030022/log.txt)

**Reply by seerge** [reply]
I see that you have set a custom High Performance power plan in a config only for one of the modes (Turbo?). For others you didn't set that, so system uses default Balanced as it should.

```
9/16/2024 8:03:52 PM: Power Plan 8c5e7fda-e8bf-4a96-9a85-a6e23a8c635c:OK <--- your turbo mode uses HIGH PERFORMANCE PLAN
9/16/2024 8:03:52 PM: Mode = 1 : OK 
```

```
9/16/2024 8:04:05 PM: Power Plan 381b4222-f694-41f0-9685-ff5bb260df2e:OK <--- CUSTOM MODE USES DEFAULT BALANCED PLAN
9/16/2024 8:04:05 PM: Power Mode 00000000-0000-0000-0000-000000000000 -> ded574b5-45a0-4f42-8737-46345c09c238:OK
9/16/2024 8:04:05 PM: Mode = 4 : OK
9/16/2024 8:04:06 PM: PowerLimit A3 = 150 : OK
9/16/2024 8:04:06 PM: PowerLimit A0 = 150 : OK
```

If you want to use custom power plans, you need to set it config for each mode you use as I have tried to explain multiple times above.

**Reply by seerge** [reply]
I can only imagine you are trying to customize mode called "MAX+" on your screenshot? It's a 4th mode in a row so you need to add to `config.json` following line (3 is cause numeration of asus modes starts from 0)

```
  "scheme_3": "8c5e7fda-e8bf-4a96-9a85-a6e23a8c635c",
```

**Reply by Azzkaryon** [reply]
Yeah i think i need to wait for him to do it:))

**Reply by jurnin** [reply]
Yes we got it now, I just didn't realize i needed a scheme 3, problem solved now thanks for helping and your patience <3 awesome tool you've made

**Comment by Psioo2** [comment]
Hello, for anyone having issues, for me scheme override didn't work for turbo, but worked for other modes. It defaulted to high performance anyway. I had to change the "powermode_1" (that contained GUID of the powermode that turbo mode defaulted to) to all zeroes.
BEFORE:
```
  "powermode_1": "8c5e7fda-e8bf-4a96-9a85-a6e23a8c635c",
  "scheme_0": "f2f9f02e-64a5-49c5-b8af-f498e5ce6e07",
  "scheme_1": "f2f9f02e-64a5-49c5-b8af-f498e5ce6e07",
  "scheme_2": "f2f9f02e-64a5-49c5-b8af-f498e5ce6e07",
```
AFTER:
```
  "powermode_1": "00000000-0000-0000-0000-000000000000",
  "scheme_0": "f2f9f02e-64a5-49c5-b8af-f498e5ce6e07",
  "scheme_1": "f2f9f02e-64a5-49c5-b8af-f498e5ce6e07",
  "scheme_2": "f2f9f02e-64a5-49c5-b8af-f498e5ce6e07",
```

Also I swear the "powermode_1" thingy didn't exist before I started messing with the config.
Here's my logs just in case:
[log.txt](https://github.com/user-attachments/files/20413130/log.txt)
You can see me changing the "powerplan_1" thing in 17:46

**Reply by seerge** [reply]
@Psioo2 because "High Performance Plan" is actually not really a Power Mode but Power Plan (added mainly to simplify life for people who still want to use it). If you want to use custom plan GUID - you better set it to Best Performance (or anything else) :)
